test/BodyParams/BodyParamsMiddlewareTest.php

```diff
@@ -52,6 +51,7 @@ public function setUp(): void
         $this->body->rewind();
     }
 
+    /** @return RequestHandlerInterface|ObjectProphecy */
```

Comment:
In cases like this, it's good to also include:

```php
@psalm-return RequestHandlerInterface&ObjectProphecy
```

as it will reduce the number of errors Psalm reports.

(Do this throughout - I don't want to open a comment for each and every one!)

Reply:
For the return types, I dropped prophecy here in favour of PHPUnit mocks because adding the annotations caused more problems that they solved and my psalm-fu is not exactly black-belt standard